package.json

```diff
@@ -2827,6 +2827,212 @@
+														"type": "command",
+														"command": "editor.action.formatDocument.multiple"
+													},
+													{
+														"key": "S",
+														"name": "+Format selection with formatter",
+														"type": "command",
+														"command": "editor.action.formatSelection.multiple"
+													}
+												]
+											},
+											{
+												"key": "d",
+												"name": "+Debug",
+												"type": "bindings",
+												"bindings": [
+													{
+														"key": "l",
+														"name": "List process for attach",
+														"type": "command",
+														"command": "csharp.listProcess"
+													},
+													{
+														"key": "L",
+														"name": "List remote processes for attach",
+														"type": "command",
+														"command": "csharp.listRemoteProcess"
+													}
+												]
+											},
+											{
+												"key": "g",
+												"name": "+Go to",
+												"type": "bindings",
+												"bindings": [
+													{
+														"key": "d",
+														"name": "Go to definition",
+														"type": "command",
+														"command": "editor.action.revealDefinition"
+													},
+													{
+														"key": "e",
+														"name": "Go to errors/problems",
+														"type": "command",
+														"command": "workbench.action.problems.focus"
+													},
+													{
+														"key": "g",
+														"name": "Go to definition",
+														"type": "command",
+														"command": "editor.action.revealDefinition"
+													},
+													{
+														"key": "i",
+														"name": "Go to implementations",
+														"type": "command",
+														"command": "editor.action.goToImplementation"
+													},
+													{
+														"key": "r",
+														"name": "Peek references",
+														"type": "command",
+														"command": "editor.action.referenceSearch.trigger"
+													},
+													{
+														"key": "D",
+														"name": "Peek definition",
+														"type": "command",
+														"command": "editor.action.peekDefinition"
+													},
+													{
+														"key": "I",
+														"name": "Find all implementation",
+														"type": "command",
+														"command": "references-view.findImplementations"
+													},
+													{
+														"key": "R",
+														"name": "Find all references",
+														"type": "command",
+														"command": "references-view.findReferences"
+													}
+												]
+											},
+											{
+												"key": "r",
+												"name": "+Refactor",
+												"type": "bindings",
+												"bindings": [
+													{
+														"key": ".",
+														"name": "Quick fix",
+														"type": "command",
+														"command": "editor.action.quickFix"
+													},
```

Comment:
isn't this `spc a a` in spacemacs?
https://develop.spacemacs.org/layers/+tools/lsp/README.html

If so, we should add this to all major modes maybe

Reply:
I was borrowing this concept from the [c++ layer](https://develop.spacemacs.org/layers/+lang/c-c++/README.html#refactor). It makes more sense to be consistent with lsp layer since all languages in vscode is going to use LSP. Should we keep `SPC m r .`?

SPC m a a | Execute code action | `editor.action.codeAction`
SPC m a f | Execute fix action | `editor.action.quickFix`
SPC m a r | Execute refactor action | `editor.action.refactor`
SPC m a s | Execute source action |  `editor.action.sourceAction`

We should open another issue to track this. Probably note that we need to add this table to the major mode contribution guide as well like the format table.

Reply:
opened #181 and created https://github.com/VSpaceCode/vspacecode.github.io/pull/41
Please add more details to the issue or the PR if you think it's useful.
`spc m r .` does not make a lot of sense to me, since it is not related to refactoring, so ideally I would remove it.
By the way, yes, we can merge the PR as it is, since `spc m a` could be added later.

Reply:
Thanks for creating the issue :)

> spc m r . does not make a lot of sense to me, since it is not related to refactoring, so ideally I would remove it.
By the way, yes, we can merge the PR as it is, since spc m a could be added later.

It contains code actions which have refactoring options?

Reply:
Since we have an issue tracking this, resolved this comment.